packages/gatsby/src/utils/start-server.ts

```diff
@@ -427,20 +382,6 @@ module.exports = {
     }
   )
 
-  // Disable directory indexing i.e. serving index.html from a directory.
-  // This can lead to serving stale html files during development.
-  //
-  // We serve by default an empty index.html that sets up the dev environment.
-  app.use(developStatic(`public`, { index: false }))
```

Comment:
Is moving this below is actually needed for this PR? (Just trying to understand if this is critical for this PR and I'm missing something)

Reply:
Yeah this was necessary cause page-data files are being written to disk and thus developStatic prevents from hitting /page-data